jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/ResponseHttpFields.java

```diff
@@ -113,15 +129,19 @@ public boolean hasNext()
             @Override
             public HttpField next()
             {
-                return i.next();
+                _current = i.next();
+                return _current;
             }
 
             @Override
             public void remove()
             {
                 if (_committed.get())
                     throw new UnsupportedOperationException("Read Only");
+                if (isPersistent(_current))
+                    throw new IllegalStateException("Persistent field");
```

Comment:
My interpretation of the `Iterator.remove()` javadoc makes me think `UnsupportedOperationException` should be thrown here.